File: example/ios/Runner/StarPrinter/TextReceipt.m
//
//  TextReceipt.m
//  StarPrinter
//
//  Created by Shailesh Lolam on 11/14/17.
//  Copyright © 2017 Shailesh Lolam. All rights reserved.
//

#import <Foundation/Foundation.h>
#import <StarIO_Extension/StarIoExt.h>
#import "TextReceipt.h"


@implementation TextReceipt

SCBAlignmentPosition alignment;

-(instancetype) initWithLanguage: (Languages) language AndSize: (PaperSizes) paperSize{
    self.language = language;
    self.paperSize = paperSize;
    self.builder = [StarIoExt createCommandBuilder: StarIoExtEmulationStarLine];
    alignment = SCBAlignmentPositionLeft;
    return self;
}

-(void) addAlignLeft{
    alignment = SCBAlignmentPositionLeft;
    [self.builder appendAlignment:SCBAlignmentPositionLeft];
}
-(void) addAlignCenter{
    alignment = SCBAlignmentPositionCenter;
    [self.builder appendAlignment:SCBAlignmentPositionCenter];
}
-(void) addAlignRight{
    alignment = SCBAlignmentPositionRight;
    [self.builder appendAlignment:SCBAlignmentPositionRight];
}

-(void) setBlackColor{
    [self.builder appendBytes:"\x1b\x35" length:sizeof("\x1b\x34") - 1];
}

-(void) setRedColor{
    [self.builder appendBytes:"\x1b\x34" length:sizeof("\x1b\x34") - 1];
}

-(void) addText: (NSString *) string{
    [self.builder appendData:[string dataUsingEncoding:NSUTF8StringEncoding]];
}

-(void) addDoubleText: (NSString *) string{
    [self.builder appendDataWithMultiple:[string dataUsingEncoding:NSUTF8StringEncoding] width:2 height:2];
}

-(void) addBoldText: (NSString *) string{
    [self.builder appendDataWithEmphasis:[string dataUsingEncoding:NSUTF8StringEncoding]];
}

-(void) addUnderlinedText: (NSString *) string{
    [self.builder appendDataWithUnderLine:[string dataUsingEncoding:NSUTF8StringEncoding]];
}

-(void) addInverseText: (NSString *) string{
    [self.builder appendDataWithInvert:[string dataUsingEncoding:NSUTF8StringEncoding]];
}

//-(void) addBarcode: (NSString *) string  height: (int) height{
//    [self.builder appendBarcodeData:[string dataUsingEncoding:NSASCIIStringEncoding] symbology:SCBBarcodeSymbologyCode128 width:SCBBarcodeWidthMode2 height:height hri:YES];
//}

-(void) cutPaper{
    [self.builder appendCutPaper:SCBCutPaperActionPartialCutWithFeed];
}

@end

File: android/src/main/java/com/getzuza/starprinter/starprinter/TextReceipt.java
package com.getzuza.starprinter.starprinter;

import android.content.Context;

import com.starmicronics.starioextension.ICommandBuilder;
import com.starmicronics.starioextension.StarIoExt;

public class TextReceipt extends Receipt  {

    private ICommandBuilder.AlignmentPosition _alighment;

    TextReceipt(Context context, Languages language, PaperSizes paperSize) {
        _context = context;
        _languages = language;
        _size = paperSize;
        _builder = StarIoExt.createCommandBuilder(StarIoExt.Emulation.StarLine);
        _alighment = ICommandBuilder.AlignmentPosition.Left;
    }

    @Override
    public void addAlignLeft() {
        _alighment = ICommandBuilder.AlignmentPosition.Left;
        setAlignment();
    }

    @Override
    public void addAlignRight() {
        _alighment = ICommandBuilder.AlignmentPosition.Right;
        setAlignment();
    }

    @Override
    public void addAlignCenter() {
        _alighment = ICommandBuilder.AlignmentPosition.Center;
        setAlignment();
    }

    private void setAlignment() {
        _builder.appendAlignment(_alighment);
    }

    @Override
    public void setBlackColor() {
        _builder.append(new byte[] {0x1b,0x35});
    }

    @Override
    public void setRedColor() {
        _builder.append(new byte[] {0x1b,0x34});
    }

    @Override
    public void addText(String text) {
        _builder.append(text.getBytes());
    }

    @Override
    public void addDoubleText(String text) {
        _builder.appendMultiple(text.getBytes(), 2, 2);
    }

    @Override
    public void addBoldText(String text) {
        _builder.appendEmphasis(text.getBytes());
    }

    @Override
    public void addUnderlinedText(String text) {
        _builder.appendUnderLine(text.getBytes());
    }

    @Override
    public void addInverseText(String text) {
        _builder.appendInvert(text.getBytes());
    }
}
